WSL/disk-space.md

```diff
@@ -0,0 +1,217 @@
+      ```
+
+> [!IMPORTANT]
+> We recommend that you do not modify, move, or access the WSL related files located inside of your AppData folder using Windows tools or editors. Doing so could cause your Linux distribution to become corrupted. If you would like to access your Linux files from Windows, that is possible via the path `\\wsl$\<distroName>\`. Open your WSL distribution and enter `explorer.exe .` to view that folder. To learn more, see the blog post: [Accessing Linux files from Windows](https://devblogs.microsoft.com/commandline/whats-new-for-wsl-in-windows-10-version-1903/#accessing-linux-files-from-windows).
+
+## How to repair a VHD mounting error
+
+If you encounter an error related to "mounting the distribution disk", this could be due to a sudden shutdown or power outage and may result in the Linux distribution VHD being switched to *read-only* to avoid data loss. You can repair and restore the distribution using the `e2fsck` Linux command by following the steps below.
+
+### Use the lsblk command to identify the block device name
+
+When WSL 2 installs a Linux distribution, it is mounting the distribution as a Virtual Hard Disk (VHD) with it's own file system. Linux refers to these hard drives as "block devices" and you can view information about them by using the `lsblk` command.
+
+To find the names of the block devices currently being used by WSL 2, open your distribution and enter the command: `lsblk`. (Or open PowerShell and enter the command: `wsl.exe lsblk`.) The output will look something like this:
+
+```bash
+NAME MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
+sda    8:0    0 363.1M  1 disk
+sdb    8:16   0     8G  0 disk [SWAP]
+sdc    8:32   0   1.5T  0 disk
+sdd    8:48   0     1T  0 disk /mnt/wslg/distro
+```
+
+Information about the block device includes:
+
+- **NAME**: The name assigned to the device will be sd[a-z], referring to the SCSI Disk with a letter designation for each disk being used. `sda` is always the system distribution.
+- **MAJ:MIN**: Represents numbers used by the Linux kernel to internally identify the devices with the first number representing the device type (8 is used for Small Computer System Interface/SCSI disks, an alternative to USB).
+- **RM**: Let's us know if the device is removable (1) or not (0).
+- **SIZE**: lets us know how much space the device is using.
+- **RO**: Let's us know if the device is read-only (1) or not (0).
+- **TYPE**: Refers to the device type (disk in this case, could also be something like lvm for a logical volume device).
+- **MOUNTPOINTS**: Refers to the current directory on the files system where the block device is located (SWAP is for preconfigured inactive memory so no mountpoint).
+
+### Read-only fallback error
+
+If WSL encounters an error while installing (mounting) a Linux distribution, it may mount the distribution as read-only (RO = 1) as a fallback. If that happens, the distribution may display the following error during startup:
```

Comment:
Let's remove `installing` here distribution installation failures are unrelated to this page

Reply:
I'm a bit confused about the scenario for this then. A few lines earlier I wrote: "When WSL 2 installs a Linux distribution, it is mounting the distribution as a Virtual Hard Disk (VHD) with it's own file system." Isn't the scenario here that a WSL user is installing a new Linux distribution and encounters a mounting error? Maybe I need more clarity on the scenario in which they would encounter a mounting error... should this instead be in the "Mount a Linux disk in WSL 2" doc?

Reply:
I don't think that the average WSL user will understand what "a WSL distribution mounting it's disk" means. I don't... and would've assumed that it's related to using the wsl --mount command to mount a disk. Now that I've spent some time with this, I *think* that I'm beginning to understand that each time a user opens a Linux distro that has been installed, WSL goes through the process of "mounting" that ext4.vhd disk so that it can be used (?)... and that during that process of opening (mounting) the distro, an error might occur (due to a power failure or something?), which would result in needing to follow these Repair instructions.   
Is that correct?

Reply:
Here's the update that I made -- 
"If WSL encounters a "mounting error" when opening a Linux distribution, the distribution may be set as read-only (RO = 1) as a fallback. If that happens..."

It leaves things a bit vague as to _why_ such a mounting error might occur... but I'm guessing that there are lots of different possible reasons (?)... but the main issue being resolved here is that this sort of mounting error message would not be related to: 1) Installing a distro, nor 2) using the wsl.exe --mount command. Correct?